.travis.yml

```diff
@@ -1,83 +1,126 @@
+sudo: false
 language: python
 
-os:
-  - linux
-
-python:
-  - "pypy"
-  - "2.6"
-  - "2.7"
-  - "3.2"
-  - "3.3"
-  - "3.4"
-
 matrix:
   include:
+  - language: generic
+    os: osx
+    env: TOXENV=py27
+  - python: 2.6 # these are just to make travis's UI a bit prettier
+    env: TOXENV=py26
+  - python: 2.7
+    env: TOXENV=py27
+  - python: 3.2
+    env: TOXENV=py32
+  - python: 3.3
+    env: TOXENV=py33
+  - python: 3.4
+    env: TOXENV=py34
+  - python: pypy
+    env: TOXENV=pypy
   # Also run the tests against cryptography master.
-  - python: "2.6"
+  - python: 2.6
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - python: "2.7"
+  - python: 2.7
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - python: "3.2"
+  - python: 3.2
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - python: "3.3"
+  - python: 3.3
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - python: "3.4"
+  - python: 3.4
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - python: "pypy"
+  - python: pypy
     env:
       CRYPTOGRAPHY_GIT_MASTER=true
 
   # Also run at least a little bit against an older version of OpenSSL.
-  - python: "2.7"
+  - python: 2.7
     env:
       OPENSSL=0.9.8
+    addons:
+      apt:
+        sources:
+          - lucid
+        packages:
+          - libssl-dev/lucid
 
   # Let the cryptography master builds fail because they might be triggered by
   # cryptography changes beyond our control.
   allow_failures:
-  - env:
+  - language: generic
+    os: osx
+    env: TOXENV=py27
+  - python: 2.6
+    env:
+      CRYPTOGRAPHY_GIT_MASTER=true
+  - python: 2.7
+    env:
       CRYPTOGRAPHY_GIT_MASTER=true
-  - env:
+  - python: 3.2
+    env:
+      CRYPTOGRAPHY_GIT_MASTER=true
+  - python: 3.3
+    env:
+      CRYPTOGRAPHY_GIT_MASTER=true
+  - python: 3.4
+    env:
+      CRYPTOGRAPHY_GIT_MASTER=true
+  - python: pypy
+    env:
+      CRYPTOGRAPHY_GIT_MASTER=true
+  - python: 2.7
+    env:
       OPENSSL=0.9.8
+    addons:
```

Comment:
Can you add a comment here explaining this section needs to match the above equivalent section in the matrix?